### PR TITLE
Adding fields to Role and Task

### PR DESCRIPTION
**Pull request by fao89**
- add Role.gather_facts
- add Task.vars

### CODE REVIEW

**Review by fultonj — COMMENTED** [review]
This all looks good to me but I have one question about the type for the vars below.

File: api/bases/ansibleee.openstack.org_openstackansibleees.yaml [hunk 1]
```diff
@@ -1904,6 +1907,12 @@ spec:
                           items:
                             type: string
                           type: array
+                        vars:
+                          items:
+                            type: object
+                            x-kubernetes-preserve-unknown-fields: true
+                          type: array
```
Comment (fultonj) [inline api/bases/ansibleee.openstack.org_openstackansibleees.yaml:1914]: `ansibleVars` in dataplane-operator node was converted to a string recently [1]. Should this `vars` field be consistent with it?

[1] 
https://github.com/openstack-k8s-operators/dataplane-operator/pull/23/commits/fa4d55e520360a3cbeeaa38cbeca317b400d3cb8#diff-1988fd56b05f07f65f48a21b48d923dcd18cb5d992d6dc1a0fba805f9e4cf553L54

Comment (fao89) [reply api/bases/ansibleee.openstack.org_openstackansibleees.yaml:1914]: Nice catch! I'm gonna change it!

**Review by fultonj — COMMENTED** [review]
Looks good to me. I think we should merge it. Do we want one more pair of eyes or should I merge?

**Comment by fao89** [comment]
> Looks good to me. I think we should merge it. Do we want one more pair of eyes or should I merge?

+1 to merge

**Pull request merged**